File: Desktop/src/api/reports.ts
import { api } from './axios';

export interface SalesSummary {
  totalSales: number;
  totalTransactions: number;
  totalDiscount: number;
  totalTax: number;
}

export interface TopProduct {
  _id: string;
  name: string;
  qty: number;
  revenue: number;
}

export interface StaffPerformance {
  _id: string;
  name: string;
  totalSales: number;
  transactions: number;
}

export interface ReportParams {
  from?: string;
  to?: string;
  period?: string;
}

export const reportApi = {
  salesSummary: (params: ReportParams = {}) =>
    api
      .get<{ data: SalesSummary }>('/client/reports/sales', { params })
      .then((r) => r.data.data),

  topProducts: (params: ReportParams & { limit?: number } = {}) =>
    api
      .get<{ data: TopProduct[] }>('/client/reports/top-products', { params })
      .then((r) => r.data.data),

  staff: (params: ReportParams = {}) =>
    api
      .get<{ data: StaffPerformance[] }>('/client/reports/staff', { params })
      .then((r) => r.data.data),

  exportCsv: (params: ReportParams = {}) =>
    api
      .get('/client/reports/export', { params, responseType: 'blob' })
      .then((r) => r.data),
};

File: Android/client/lib/api/reports.dart
import 'api_client.dart';

class ReportsApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> salesSummary({
    Map<String, dynamic>? query,
  }) => _api.get('/client/reports/sales', query: query);
  static Future<Response<dynamic>> topProducts({Map<String, dynamic>? query}) =>
      _api.get('/client/reports/top-products', query: query);
  static Future<Response<dynamic>> staff({Map<String, dynamic>? query}) =>
      _api.get('/client/reports/staff', query: query);
  static Future<Response<dynamic>> exportCsv({Map<String, dynamic>? query}) =>
      _api.get('/client/reports/export', query: query);
}
